docs/developer/developer-mode-installation.md

```diff
@@ -0,0 +1,144 @@
+    cni:
+    - multus
+    - canal
+    EOF
+    ```
+
+1. Install RKE2.
+    ```bash
+    curl -sfL https://get.rke2.io | sudo sh -
+    sudo systemctl enable rke2-server.service
+    sudo systemctl start rke2-server.service
+    ```
+
+1. Install system-upgrade-controller. The component installs `upgrade.cattle.io/v1` CRDs.
+    ```bash
+    kubectl apply -f https://github.com/rancher/system-upgrade-controller/releases/download/v0.13.1/system-upgrade-controller.yaml
+    ```
+
+1. Get kubeconfig.
+    ```bash
+    sudo chmod 644 /etc/rancher/rke2/rke2.yaml
+    mkdir -p ~/.kube
+    cp /etc/rancher/rke2/rke2.yaml ~/.kube/config
+    ```
+
+1. Create `cattle-system` namespace.
+    ```bash
+    kubectl create ns cattle-system
+    ```
+
+1. Add Rancher repo.
+    ```bash
+    helm repo add rancher-latest https://releases.rancher.com/server-charts/latest
+    ```
+
+1. Install rancher v2.7.5 chart.
+    ```bash
+    helm install rancher rancher-latest/rancher \
+    --namespace cattle-system \
+    --set tls=external \
+    --set rancherImagePullPolicy=IfNotPresent \
+    --set rancherImage=rancher/rancher \
+    --set rancherImageTag=v2.7.5 \
+    --set noDefaultAdmin=false \
+    --set features="multi-cluster-management=false\,multi-cluster-management-agent=false" \
+    --set useBundledSystemChart=true \
+    --set bootstrapPassword=admin
+    ```
+
+1. Clone rancher/charts repository.
+    ```bash
+    git clone https://github.com/rancher/charts -b dev-v2.7
+    ```
+
+1. Install rancher-montering-crd chart.
+    ```bash
+    helm install rancher-monitoring-crd ./charts/charts/rancher-monitoring-crd/102.0.2+up40.1.2/
+    ```
+
+1. Create `harvester-system` namespace.
+    ```bash
+    kubectl create ns harvester-system
+    ```
+
+1. Clone harvester/harvester repository.
+    ```bash
+    git clone https://github.com/harvester/harvester.git
+    ```
+
+1. Install harvester-crd chart.
+    ```bash
+    helm install harvester-crd ./harvester/deploy/charts/harvester-crd --namespace harvester-system
+    ```
+
+1. Install harvester chart. Use static mode kube-vip for developer mode installation.
+    ```bash
+    VIP_IP="replace with your node ip, such as 192.168.5.10"
+    helm install harvester ./harvester/deploy/charts/harvester --namespace harvester-system \
+    --set harvester-node-disk-manager.enabled=true \
+    --set "harvester-node-disk-manager.labelFilter={COS_*,HARV_*}" \
+    --set harvester-network-controller.enabled=true \
+    --set harvester-network-controller.vipEnabled=true \
+    --set harvester-load-balancer.enabled=true \
+    --set kube-vip.enabled=true \
+    --set kube-vip-cloud-provider.enabled=true \
+    --set longhorn.enabled=true \
+    --set longhorn.defaultSettings.defaultDataPath=/var/lib/harvester/defaultdisk \
+    --set longhorn.defaultSettings.taintToleration=kubevirt.io/drain:NoSchedule \
+    --set service.vip.enabled=true \
+    --set service.vip.mode=static \
+    --set service.vip.ip=${VIP_IP}
```

Comment:
@starbops @w13915984028, could you share some idea about static vip mode? What is the limitation? Is it good enough for developer mode? Thank you.

Reply:
Thanks for updating the `dev mode`.

It looks fine to limit the vip to `static` in `dev mode`.

Do we also limit the `dev mode` to single-node or allow multi-node cluster?

Reply:
1. Yes, as the VIP exists, the other joining nodes should be able to contact the VIP to complete the registration flow.
2. In dev mode we don't have harvester-installer and thus mgmt-br is missing. The VIP address will be bound to the main interface, e.g., enp1s0. This is not specific to DHCP mode only.
3. To set up the cluster with DHCP mode VIP, configure it with `--set service.vip.mode=dhcp`, `--set service.vip.hwAddress=<macaddr of interest>`, and remember to to add `--set rancherEmbedded=true`, too.

Reply:
I forgot to mention the `--set rancherEmbedded=true` is also required in static mode. Otherwise, no LB Service (`ingress-expose`) will be created.

Reply:
Another thing to add: if the user is not going to set up a static DHCP lease for the VIP address beforehand, the `service.vip.hwAddress` will not be necessary. After the installation, kube-vip will acquire a random IP address from the DHCP services configured pool as the VIP address.

Reply:
@starbops, thanks for suggestions!

Yeah, from source code, we need `rancherEmbedded` to create `ingress-expose` service.

https://github.com/harvester/harvester/blob/acb6e69121a9843c8e7616de9f44401293c4d238/pkg/controller/master/rancher/register.go#L72-L99